SeaPublicWebsite.UnitTests/QuestionFlowServiceTests.cs

```diff
@@ -148,85 +148,94 @@ public void RunSkipLinkTestCases(QuestionFlowServiceTestCase testCase)
                 "EnergyEfficiency",
                 new { reference = "ABCDEFGH" }
             )),
         new(
-            "No EPC found goes back to postcode if epc count is 0",
+            "Property type goes back to confirm EPC details if epc exists with property type and age",
             new Input(
-                QuestionFlowPage.NoEpcFound,
+                QuestionFlowPage.PropertyType,
                 "ABCDEFGH",
-                epcCount: 0
+                epc: new Epc
+                {
+                    PropertyType = PropertyType.House,
+                    HouseType = HouseType.Detached,
+                    ConstructionAgeBand = HomeAge.From1950To1966
+                }
             ),
             new PathByActionArguments(
-                nameof(EnergyEfficiencyController.AskForPostcode_Get),
+                nameof(EnergyEfficiencyController.ConfirmEpcDetails_Get),
                 "EnergyEfficiency",
                 new { reference = "ABCDEFGH" }
             )),
         new(
-            "Property type goes back to confirm EPC details if not confirmed",
+            "Property type goes back to postcode if epc doesn't contain property type",
             new Input(
                 QuestionFlowPage.PropertyType,
                 "ABCDEFGH",
-                epcDetailsConfirmed: EpcDetailsConfirmed.No,
-                epcAddressConfirmed: EpcAddressConfirmed.Yes
+                epc: new Epc
+                {
+                    ConstructionAgeBand = HomeAge.From1950To1966
+                }
             ),
             new PathByActionArguments(
-                nameof(EnergyEfficiencyController.ConfirmEpcDetails_Get),
+                nameof(EnergyEfficiencyController.AskForPostcode_Get),
                 "EnergyEfficiency",
                 new { reference = "ABCDEFGH" }
             )),
         new(
-            "Property type goes back to no EPC found if details confirmed null but address confirmed",
+            "Property type goes back to postcode if epc doesn't contain property age",
             new Input(
                 QuestionFlowPage.PropertyType,
                 "ABCDEFGH",
-                epcDetailsConfirmed: null,
-                epcAddressConfirmed: EpcAddressConfirmed.No
+                epc: new Epc()
+                {
+                    PropertyType = PropertyType.House,
+                    HouseType = HouseType.Detached
+                }
             ),
             new PathByActionArguments(
-                nameof(EnergyEfficiencyController.NoEpcFound_Get),
+                nameof(EnergyEfficiencyController.AskForPostcode_Get),
                 "EnergyEfficiency",
                 new { reference = "ABCDEFGH" }
             )),
         new(
-            "Property type goes back to confirm address if address confirmed but details confirmed is null",
+            "Property type goes back to no EPC found if searched for EPC but no EPC set",
```

Comment:
does this cover the case that a user clicked "Yes" but then clicked "cancel and continue without epc? this should go back to FindEpc not to NoEpcFound I think.

Reply:
Good spot, this is broken